xgcm/padding.py

```diff
@@ -0,0 +1,306 @@
+"""
+Handle all padding.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Dict, Optional, Tuple, Union
+
+import xarray as xr
+
+if TYPE_CHECKING:
+    from .grid import Grid
+
+_XGCM_BOUNDARY_KWARG_TO_XARRAY_PAD_KWARG = {
+    "periodic": "wrap",
+    "fill": "constant",
+    "extend": "edge",
+    None: "wrap",  # current default is periodic. This should achieve that.
+}
+
+
+def _maybe_swap_dimension_names(da, from_name, to_name):
+    # renames 1D slices and swaps dimension names for higher dimensional slices
+    if to_name in da.dims:
+        da = da.rename({to_name: to_name + "dummy"})
+        if from_name in da.dims:
+            da = da.rename({from_name: to_name})
+        da = da.rename({to_name + "dummy": from_name})
+    else:
+        da = da.rename({from_name: to_name})
+    return da
+
+
+def _pad_face_connections(da, grid, padding_width, **kwargs):
+
+    # Can I get rid of all the mess with the coordinates by stripping them here?
+    # The output of the padding will not fit any coordinates anymore anyways and
+    # after the ufunc is applied they will all be back on place. So lets make our
+    # lives easier here.
+    da = da.reset_coords(drop=True).reset_index(
+        [di for di in da.dims if di in da.coords], drop=True
+    )
```

Comment:
Is this a good idea? What if the `da` has a coordinate that doesn't depend on the dimension of the axis along which we are applying the ufunc? In that case `xarray.apply_ufunc` should keep it, and if you drop all coordinates here it would change that behaviour. (Are you dropping all coordinates? I can't easily understand what this line does...)